### PR TITLE
Fix missing Trim in Roles and Schemes split

### PR DESCRIPTION
**Pull request by phenixdotnet**
PR for issue #627

### CODE REVIEW

File: src/Microsoft.AspNet.Authorization/AuthorizationPolicy.cs [hunk 1]
```diff
@@ -88,6 +88,9 @@ public static AuthorizationPolicy Combine(AuthorizationOptions options, IEnumera
                 var rolesSplit = authorizeAttribute.Roles?.Split(',');
                 if (rolesSplit != null && rolesSplit.Any())
                 {
+                    for (int i = 0; i < rolesSplit.Length; ++i)
+                        rolesSplit[i] = rolesSplit[i]?.Trim();
```
Comment (HaoK) [inline src/Microsoft.AspNet.Authorization/AuthorizationPolicy.cs]: nit: coding style requires { } even around one liners, same for below

File: src/Microsoft.AspNet.Authorization/AuthorizationPolicy.cs [hunk 2]
```diff
@@ -96,7 +99,9 @@ public static AuthorizationPolicy Combine(AuthorizationOptions options, IEnumera
                 {
                     foreach (var authType in authTypesSplit)
                     {
-                        policyBuilder.AuthenticationSchemes.Add(authType);
+                        if (string.IsNullOrEmpty(authType))
```
Comment (HaoK) [inline src/Microsoft.AspNet.Authorization/AuthorizationPolicy.cs]: Instead of this explicit check for empty, can you use the the overoad of Split which takes in `StringSplitOptions.RemoveEmptyEntries`

File: test/Microsoft.AspNet.Authorization.Test/AuthorizationPolicyFacts.cs [hunk 3]
```diff
@@ -67,5 +67,39 @@ public void CanReplaceDefaultPolicy()
...
+            var combined = AuthorizationPolicy.Combine(options, attributes);
+
+            Assert.True(combined.Requirements.Any(r => r is RolesAuthorizationRequirement));
+            var rolesAuthorizationRequirement = combined.Requirements.OfType<RolesAuthorizationRequirement>().First();
+            Assert.Equal(2, rolesAuthorizationRequirement.AllowedRoles.Count());
+            Assert.True(rolesAuthorizationRequirement.AllowedRoles.Any(r => r.Equals("r1")));
+            Assert.True(rolesAuthorizationRequirement.AllowedRoles.Any(r => r.Equals("r2")));
+        }
+
+        [Fact]
```
Comment (HaoK) [inline test/Microsoft.AspNet.Authorization.Test/AuthorizationPolicyFacts.cs]: Can you also add/or update one of these test for the an empty scheme/role?
Roles = "r1,,,,r2", AUthenticationSchemes = "a1,,,a2"

**Comment by phenixdotnet** [comment]
I added 2 tests with the roles or authentication scheme empty or with white space.
I have to keep the ```if(string.IsNullOrWhiteSpace())```and use the ```StringSplitOptions.RemoveEmptyEntries``` to remove the empty or white space entries.

**Comment by HaoK** [comment]
Okay so if we still need the NullOrWhiteSpace check, there's no real value in the RemoveEmptyEntries option, sorry about that, can you revert that back to the simple Split and other than that this looks good, I'll merge this in next week...  Thanks!

**Comment by phenixdotnet** [comment]
Ok no prob :).

I will do that ASAP and let you known.

**Comment by phenixdotnet** [comment]
Rework done.
One last things, I used linq to filter roles (````roles.Where(r => !string.IsNullOrWhiteSpace(r)).Select(r => r => r.Trim());````)
Not sure if it's the best way to do this.

**Comment by HaoK** [comment]
Looks good, I'll get this in next week, thanks again!

**Comment by phenixdotnet** [comment]
You're welcome :)
